src/Pact/Runtime/Capabilities.hs

```diff
@@ -174,17 +183,27 @@ checkManaged applyF cap = use (evalCapabilities . capManaged) >>= go
           Just (Right newMgdCap) ->
             return (Just newMgdCap,[],S.insert newMgdCap ms)
 
-    runManaged mcs = case _csScope mcs of -- validate scope
-      CapManaged (Just mf) -> case _csCap mcs of -- validate user mg cap and has mgr fun
-        UserCapability mqn mas -> case cap of -- validate user test cap
-          UserCapability cqn cas
-            | cqn == mqn -> -- check type match and apply mgr fun
-                Just <$> applyMgrFun mcs mqn mf mas cas
-            | otherwise -> noMatch
-          _ -> noMatch
-        _ -> noMatch
+    runManaged mcs = case (_csCap mcs,cap) of -- validate mgr fun, mgr cap, acquired cap
+      (UserCapability mqn mas,UserCapability cqn cas) -- managed user cap w/ fun
+        | cqn == mqn ->
+            -- apply mgr fun on cap domain match
+            Just <$> applyMgrFun mcs mqn mgrFun mas cas
+
+        | otherwise -> noMatch
+      (ModuleAdminCapability mmn,ModuleAdminCapability cmn)
+        -- TODO, this is unreachable, as module admin has 'Nothing' for mgrFun.
+        -- Leaving this in as there is no way currently to acquire admin (as opposed
+        -- to install, which is what 'acquireModuleAdmin' does, which always
+        -- succeeds with the "previously installed" check in 'acquireCapability'.
+        -- Leaving it here for the semantics, and noting that we need a better
+        -- rep of the mgrFun to represent admin if we implement acquire.
+        | mmn == cmn -> -- admin trivially succeeds on matching module
+          return $ Just (Right mcs)
+        | otherwise -> noMatch
```

Comment:
This seems like something we shouldn't be coding ourselves into, but it's safe for now. Let's revisit when we have time to optimize